code.ts

```diff
@@ -164,21 +186,24 @@ async function startTimer(node: TextNode, seconds: number, template: string, sta
     };
 
     // checking if pause was NOT clicked
-    if (!pause) {
-      if (secondsToGo > 0) {
-        newText = fillUpTimeStringWithTemplate(secondsToInterval(secondsToGo), template);
-        if (startsWithTimer) {
-          newText = "Timer: " + newText;
+    if (!reset) {
+      if (!pause) {
+        if (secondsToGo > 0) {
+          newText = fillUpTimeStringWithTemplate(secondsToInterval(secondsToGo), template);
+          figma.ui.postMessage(["counting", newText, timerID, secondsToGo, seconds]);
```

Comment:
would be nice to move the message posting into a function that takes state, text, id and secondsToGo as an argument

Reply:
can you elaborate, not sure if I understand @lennet

Reply:
The current implementation works fine but it is an easy spot to introduce bugs. So a safer option would hiding the Figma.ui.postMessage behind a function that takes the arguments

Reply:
done, maybe check if that's what you imagined